docs/uml/puml/Parser/Parser.puml

```diff
@@ -0,0 +1,80 @@
+@startuml
+!include ../.PlantUMLClass.config
+class Parser {
+    + getUserCommand(userInput: String): Command
+    - processCatalogCommand(userInput: String): Command
+    - processAuthorMangaCommand(userInput: String, isDelete: boolean): Command
+    - isValidDeleteCommand(userInput:String): boolean
+    - processViewCommand(userInput: String): Command
+    - processSalesCommand(userInput: String): Command
+}
+
+Parser ..> ArgumentFinder : extracts arguments using >
+
+abstract class ArgumentFinder <<abstract>> {
+    # pattern: Pattern
+    # matcher: Matcher
+    # option: String
+    + ArgumentFinder(pattern: Pattern, option: String): ArgumentFinder
+    + <<abstract>> getArgumentResult(userInput: String): ArgumentResult
+}
+
+ArgumentFinder ..> ArgumentResult : generates >
+ArgumentResult ..> Parser : provides arguments for >
+
+class Regex {
+    + {static} AUTHOR_NAME_EXTRACTOR_PATTERN: Pattern
+    + {static} MANGA_NAME_EXTRACTOR_PATTERN: Pattern
+    + {static} PRICE_EXTRACTOR_PATTERN: Pattern
+    + {static} QUANTITY_EXTRACTOR_PATTERN: Pattern
+    - {static} generateExtractorPattern(includedOptionFlag: String): Pattern
+}
```

Comment:
Would it be better to show that `Regex` is a utility class with `<<utility>>`, since it only has `static` attributes and methods? See `docs/uml/puml/StorageClass/StorageClass.puml`: 
![image](https://github.com/user-attachments/assets/afe96bd3-c2cd-4b84-8a08-c0c9e9af96f2)